napari/layers/labels/labels.py

```diff
@@ -665,6 +679,35 @@ def fill(self, coord, new_label, refresh=True):
         if refresh is True:
             self.refresh()
 
+    @lru_cache(maxsize=64)
+    def sphere_indices(self, radius, sphere_dims):
```

Comment:
This should be function in `_labels_utils.py` that then gets its own tests in `_testst/test_labels_utils.py` and then be imported in here and used rather than being on the object itself.

Ideally we try and keep methods unlikely to be directly used by the user off the main object, and have as simple functions as we can in our utils so they can be tested in isolation

Reply:
Got it!

I'm also realizing I mucked up the tests, since I wasn't really testing the mouse interactions.

I should have put all those new tests in `test_labels.py`, right?

Or at least only add one new circle test to `test_labels_mouse_bindings.py`.

Reply:
for tests it would be nice to use a parameterized test where you move over the brush - i.e. from `circle` to `square`, see this example here https://github.com/napari/napari/blob/master/napari/_tests/test_dtypes.py#L4-L20 that will reduce some of the code duplication.

Reply:
actually that might be hard as you then need to pass the shape of what things need to look like too, maybe @jni can advise here, i'm signing off for the night

Reply:
I _think_ that should be doable if the test is designed for it. 

I've been lazy with the circle tests and just checked if there are the expected number of painted voxels/pixels for each label. It doesn't actually check the location. 

But we can parametrize the square and circle paintbrushes, and then verify that the number of labeled pixels in specific volumes are appropriate for a square or circle brush.

So our expected values for the param will just be two lists of values